### ISSUE TITLE
NZ Law Style: parallel citation issue

### ISSUE THREAD

**Issue by peterckelly** [Juris-M/zotero#97]
NZ Law Style provides its parallel citation (neutral citation) rules at https://www.lawfoundation.org.nz/style-guide2019/chapter-pt.3.3.html but in practice the neutral citation is often followed by the relevant proprietary citation (separated with a comma). Just for added complexity the court is not added in brackets in this case as the neutral citation indicates the court (High Court in this case.)

For the two citations in the attached, it should render as:

Marlborough District Council v Zindia Ltd [2019] NZHC 2765, [2020] NZRMA 216.

Assuming that this is a self-help issue, any pointers on where to start?

**Comment by peterckelly** [comment]
The RDF for these two citations is:

<rdf:RDF
 xmlns:rdf="http://www.w3.org/1999/02/22-rdf-syntax-ns#"
 xmlns:z="http://www.zotero.org/namespaces/export#"
 xmlns:dcterms="http://purl.org/dc/terms/"
 xmlns:dc="http://purl.org/dc/elements/1.1/"
 xmlns:bib="http://purl.org/net/biblio#">
    <bib:Document rdf:about="#item_2384">
        <z:itemType>case</z:itemType>
        <dcterms:isPartOf>
           <bib:CourtReporter><dc:title>NZRMA</dc:title></bib:CourtReporter>
        </dcterms:isPartOf>
        <dc:relation rdf:resource="#item_2385"/>
        <bib:pages>216</bib:pages>
        <dc:title>Marlborough District Council v Zindia Ltd</dc:title>
        <z:jurisdiction>002nzNew Zealand|NZ</z:jurisdiction>
        <z:yearAsVolume>2020</z:yearAsVolume>
        <z:displayTitle>Marlborough District Council v Zindia Ltd (NZRMA)</z:displayTitle>
    </bib:Document>
    <bib:Document rdf:about="#item_2385">
        <z:itemType>case</z:itemType>
        <dcterms:isPartOf>
           <bib:CourtReporter><dc:title>NZHC</dc:title></bib:CourtReporter>
        </dcterms:isPartOf>
        <dc:relation rdf:resource="#item_2384"/>
        <bib:pages>2765</bib:pages>
        <dc:title>Marlborough District Council v Zindia Ltd</dc:title>
        <z:jurisdiction>002nzNew Zealand|NZ</z:jurisdiction>
        <z:yearAsVolume>2019</z:yearAsVolume>
        <z:displayTitle>Marlborough District Council v Zindia Ltd (NZHC)</z:displayTitle>
    </bib:Document>
</rdf:RDF>

**Comment by fbennett** [comment]
Hello Peter! We can do this---similar patterns were addressed in the push to implement the Indigo Book. I'll take a look at the state of NZLS today, and see about migrating the style to the new modular infrastructure.

**Comment by peterckelly** [comment]
That's very exciting. I have a bunch of smaller issues which I hope I can patch once it is migrated over. There's also a research librarian at Victoria University of Wellington who is interested in helping out. I with 300 open source programmers, so I would hope I can contribute a few things before I finish my legal studies in another year...

**Comment by fbennett** [comment]
I checked, and NZ Law is already cast as a modular style, so the adjustment that you need should be easy enough to do. Jurism doesn't import RDF very well. If it's not too much trouble, could you export your test items as CSL JSON and post them here?

**Comment by peterckelly** [comment]
Thanks! I did make a tiny tweak, which was to put in the court for the NZRMA citation, which oddly also changes the jurisdiction. Feel free to homogenise the jurisdiction and remove the authority if that makes things easier for this case.
[
	{
		"type": "legal_case",
		"multi": {
			"main": {},
			"_keys": {}
		},
		"authority": "high.court",
		"collection-number": "2020",
		"container-title": "NZRMA",
		"page": "216",
		"title": "Marlborough District Council v Zindia Ltd",
		"jurisdiction": "nz:blenheim"
	},
	{
		"type": "legal_case",
		"multi": {
			"main": {},
			"_keys": {}
		},
		"collection-number": "2019",
		"container-title": "NZHC",
		"page": "2765",
		"title": "Marlborough District Council v Zindia Ltd",
		"jurisdiction": "nz"
	}
]

**Comment by peterckelly** [comment]
I do wonder if somehow I am not applying the style correctly too. For example, I see that the <terms> section of the style defines the section symbol correctly as s, but it is still popping up as § for me. (I am assuming we are talking about juris-nz.csl)

**Comment by fbennett** [comment]
As we head into a quiet holiday season ... is there any way I can help out with NZ Law and the new jurisdiction architecture?

**Comment by peterckelly** [comment]
Certainly! I have grand plans, but a little shove to get going would really help.

My grand plans involve:
- Catalyst IT, an 300-developer open source development company that I'm the legal manager for, has agreed that I can spend some testing and developer hours on the project
- I've engaged with my university's law librarians and they are keen to help, and other NZ law librarians are excited too
- I've got an initial validation by the law librarian of a set of references against NZ Law style (we call it NZLSG), from my most recent paper
- Next step is to compare those validated references to the generated ones from Jurism and raise tickets for the distinct issues
- Those could then be worked on by me, or anyone, including with those developer hours discussed above

But the key thing is to get any confidence that I can do anything!  If you would be able to make the most trivial change first, which is just to select NZ Law Style and see whether a section number in a statute comes out as "s" or “§”, and if the latter then make the change in the style to make it the former, that would help me see how things work. To my untutored eyes it's not obvious what's wrong and why I get “§”!

Happy to have a call to discuss if that's helpful. You could DM me on Twitter for my number.

Thanks!

**Comment by fbennett** [comment]
Dear Peter,

This is all very good to hear. Julia Caldwell's work on the New Zealand Law
Style with John Prebble's backing gave an early boost to Jurism
development, back when I was being stonewalled by the Bluebook and the ALWD
editors, and when the Jurism style infrastructure changed, I regretted
leaving the NZL style in the lurch. The architecture is stable now, and
work will not be wasted.

I've pushed some changes to the current copy of jm-new-zealand-law.csl. One
commit makes some changes needed to make the code validate against the
current language schema. A second makes the changes needed to change “§” to
"s" (by replacing the form="symbol" parameter on labels to form="short").
Here is that second commit:

https://github.com/Juris-M/jm-styles/commit/87afd99db364af9af5e63642d11768d863086965

The output it produces for a statute reference has a bunch of issues. Given
the many streamlining changes to the CSL-M dialect of CSL, it might be
worth considering a full rewrite of at least parts of the style code,
rather than chasing individual errata. We can chat about that, I'll send a
note to Twitter DM shortly.

One initial question is whether you are familiar with the Jurism
development tools. There is a handy test runner that can take a lot of the
pain out of style editing. That, the modular style infrastructure, and the
source/compile cycle for working with standard abbreviations are the main
bits of the learning curve. Once you have those, the rest of style design
is pretty much a lego-blocks exercise.

Will send a DM soon.

Frank
